src/ai/backend/manager/models/group.py

```diff
@@ -549,33 +568,40 @@ async def mutate(
         set_if_set(props, data, "resource_policy")
         set_if_set(props, data, "container_registry")
 
+        set_if_set(props, user_data, "users_to_add", clean_func=set)
+        set_if_set(props, user_data, "users_to_remove", clean_func=set)
+
         if "name" in data and _rx_slug.search(data["name"]) is None:
             raise ValueError("invalid name format. slug format required.")
-        if props.user_update_mode not in (None, Undefined, "add", "remove"):
-            raise ValueError("invalid user_update_mode")
-        if not props.user_uuids:
-            props.user_update_mode = None
-        if not data and props.user_update_mode is None:
+        if not data and not user_data:
             return cls(ok=False, msg="nothing to update", group=None)
 
         async def _do_mutate() -> ModifyGroup:
-            async with graph_ctx.db.begin() as conn:
-                # TODO: refactor user addition/removal in groups as separate mutations
-                #       (to apply since 21.09)
-                if props.user_update_mode == "add":
-                    values = [{"user_id": uuid, "group_id": gid} for uuid in props.user_uuids]
-                    await conn.execute(
+            async with graph_ctx.db.begin_session() as db_session:
+                users_to_add = user_data.get("users_to_add") or set()
+                users_to_remove = user_data.get("users_to_remove") or set()
```

Comment:
To make this more Pythonic:
```suggestion
                users_to_add = user_data.get("users_to_add", set())
                users_to_remove = user_data.get("users_to_remove", set())
```

Reply:
I thought that `user_data` would have "users_to_add(or users_to_remove)" key with `None` or `Undefined` value.
I updated this code and pushed